ldelf/ta_elf.c

```diff
@@ -192,22 +192,32 @@ static void save_hashtab_from_segment(struct ta_elf *elf, unsigned int type,
 	}
 }
 
-static bool check_addr_size(struct ta_elf *elf, void *addr, size_t sz)
+static void check_hashtab(struct ta_elf *elf, void *ptr, size_t num_buckets,
+			  size_t num_chains)
 {
+	size_t num_words = 2;
```

Comment:
Why does `num_words` start at `2`? To handle the initial `0`, `0` case?

Reply:
It's part of the ELF specification:
http://www.sco.com/developers/gabi/latest/ch5.dynamic.html#hash
The first two words are mandatory and when added together with 2 gives the total number of words in the hashtab.

Reply:
Maybe add a short inline commit as it's not obvious.